### PR TITLE
[Fiber photometry] Add fiber photometry time alignment

### PR DESCRIPTION
**Pull request by weiglszonja**
## Alignment

 We are aligning the starting time of the fiber photometry, video and DLC interfaces to the Bpod interface.

 We are computing the time shift from the Bpod trial start time to the Doric trial start time.

 For example, the computed time shift for this session:
 ```python
 time_shift = bpod_trial_start_times[0] - doric_trial_start_times[0]
 >>> -5.6901600000000006
 ```

 We are applying this time_shift to the timestamps for the raw fluorescence signals as:

 ```python
 doric_timestamps = doric_file["/DataAcquisition/FPConsole/Signals/Series0001/AnalogIn/Time"][:]
 aligned_timestamps = doric_timestamps + time_shift
 ```

 1) When the time shift is negative and the first aligned timestamp of the doric trace is negative:
 - shift back bpod (from every column that has a timestamp they have to be shifted back)
 - shift back session start time
 - don't have to move doric or video
 2) When the time shift is negative and the first aligned timestamp of the doric trace is positive
 - we move the doric and video backward
 3) When time shift is positive
 - we move the doric and video forward

### CODE REVIEW

**Comment by weiglszonja** [comment]
Example [file](https://neurosift.app/?p=/nwb&url=https://api.dandiarchive.org/api/assets/1ab68ea7-7f3d-4510-94b6-8a22fac2d07a/download/&dandisetId=001265&dandisetVersion=draft) from the embargo

File: src/constantinople_lab_to_nwb/fiber_photometry/fiber_photometry_notes.md [hunk 1]
```diff
@@ -128,19 +125,109 @@ For each `FiberPhotometryResponseSeries` that we add to NWB, we need to specify
...
+```
+
+```python
+doric_trial_start_times[:7]
+>>> [17.11626, 110.21736, 151.702835, 209.255035, 217.41393499999998, 221.01406, 229.73321]
+```
+
+## Alignment
+
+We are aligning the starting time of the fiber photometry, video and DLC interfaces to the Bpod interface.
```
Comment (alessandratrapani) [inline src/constantinople_lab_to_nwb/fiber_photometry/fiber_photometry_notes.md:208]:
```suggestion
We are aligning the starting time of the fiber photometry (Doric), video and DLC interfaces to the Bpod interface.
```

File: src/constantinople_lab_to_nwb/fiber_photometry/fiber_photometry_nwbconverter.py [hunk 2]
```diff
@@ -20,3 +23,102 @@ class FiberPhotometryNWBConverter(NWBConverter):
...
+            digital_stream_names = ["DI/O-1", "DI/O-2"]
+            ttl_signals = fiber_photometry_interface._get_traces(channel_column_names=digital_stream_names)
+            timestamps = fiber_photometry_interface.get_original_timestamps()
+
+        raw_behavior_interface = self.data_interface_objects["Behavior"]
+        trial_start_times_from_bpod, trial_stop_times_from_bpod = raw_behavior_interface.get_trial_times()
+
+        # we are aligning the ttl signals to the raw bpod trial start times
+        rising_frames_from_center_port_ttl = get_rising_frames_from_ttl(ttl_signals[:, 1])
+        center_port_aligned_times = timestamps[rising_frames_from_center_port_ttl]
```
Comment (alessandratrapani) [inline src/constantinople_lab_to_nwb/fiber_photometry/fiber_photometry_nwbconverter.py:44]: I would move this before line 39, but it's just my preference

File: src/constantinople_lab_to_nwb/fiber_photometry/fiber_photometry_nwbconverter.py [hunk 3]
```diff
@@ -20,3 +23,102 @@ class FiberPhotometryNWBConverter(NWBConverter):
...
+        time_shift = trial_start_times_from_bpod[0] - center_port_aligned_times[0]
+        self._time_shift = time_shift
+
+        has_deep_lab_cut = "DeepLabCut" in self.data_interface_objects
+        has_video = "Video" in self.data_interface_objects
+
+        rising_frames_from_camera_ttl = get_rising_frames_from_ttl(ttl_signals[:, 0])
+        timestamps_from_camera_ttl = timestamps[rising_frames_from_camera_ttl]
+
+        # When the time_shift is negative and the first aligned timestamp of the doric trace is negative:
```
Comment (alessandratrapani) [inline src/constantinople_lab_to_nwb/fiber_photometry/fiber_photometry_nwbconverter.py:78]: where are you checking that the time shift is negative?

File: src/constantinople_lab_to_nwb/fiber_photometry/fiber_photometry_nwbconverter.py [hunk 4]
```diff
@@ -20,3 +23,102 @@ class FiberPhotometryNWBConverter(NWBConverter):
...
+        if timestamps[0] + self._time_shift < 0:
```
Comment (weiglszonja) [inline src/constantinople_lab_to_nwb/fiber_photometry/fiber_photometry_nwbconverter.py:79]: @alessandratrapani 
isn't it this clause?

**Review by alessandratrapani — APPROVED** [review]
This is great!

**Pull request merged**